src/ONYXKEYS.ts

```diff
@@ -230,4 +232,30 @@ export default {
 
     // Report ID of the last report the user viewed as anonymous user
     LAST_OPENED_PUBLIC_ROOM_ID: 'lastOpenedPublicRoomID',
+} as const;
+
+type OnyxKeysMap = typeof ONYXKEYS;
+
+type RegularOnyxKeys = {
+    [K in ValueOf<Omit<OnyxKeysMap, 'COLLECTION' | 'FORMS'>>]: string;
 };
+
+type CollectionOnyxKeys = {
+    [K in `${ValueOf<OnyxKeysMap['COLLECTION']>}${string}`]: string;
+};
+
+type FormsOnyxKeys = {
+    [K in ValueOf<OnyxKeysMap['FORMS']>]: string;
+};
+
+type OnyxKeys = RegularOnyxKeys & CollectionOnyxKeys & FormsOnyxKeys;
```

Comment:
Let's change this part as @hayata-suenaga suggested:
```suggestion
type OnyxKeysMap = typeof ONYXKEYS;
type CollectionKey = `${ValueOf<OnyxKeysMap["COLLECTION"]>}${string}`;
type OnyxKey = DeepValueOf<Omit<OnyxKeysMap, "COLLECTION">> | CollectionKey;
```

Reply:
And `global.d.ts` should be:
```ts
// global.d.ts, E/App
declare module 'react-native-onyx' {
    interface CustomTypeOptions {
        key: OnyxKey;
        values: OnyxValues;
    }
}
```

Reply:
@blazejkustra it works! Will update the POC